cabal.haskell-ci

```diff
@@ -1,6 +1,8 @@
+distribution:           bionic
```

Comment:
Ooh, I didn't realize this was configurable now. Nice!

Out of curiosity, what is the default setting? Is there a reason to prefer something over the default setting?

Reply:
Default is xenial, bionic isn’t officially supported. Installing hvr’s ppa GHCJS on xenial however goes bad as nodejs is old there

Reply:
reverted back to `xenial` by manually installing nodejs.

Reply:
Cool! Now that we're back on Xenial, doesn't this mean we can remove the `sudo: required` bits?

Reply:
No, we cannot. `sudo apt-get install ghcjs` still needs `sudo`.

Reply:
I'm confused. https://github.com/ekmett/lens/pull/879#discussion_r310676432 gave me the impression that the only reason we needed this `sudo` business in the first place was due to the use of Bionic. Now that we're back to Xenial, shouldn't we be able to just put this stuff back into the `addons` stanza of the configuration matrix?

Reply:
No. `hvr-ghcjs` neither `nodejs` APT repositories aren't on the whitelist, and there is no way to put them there at the moment: https://github.com/travis-ci/apt-source-safelist

> At this moment, we are unable to continue processing the APT sources requests. If you require a new source, please use `sudo: required` in your `.travis.yml` file.

Reply:
Also compare
- https://travis-ci.org/ekmett/lens/builds/567513619
- https://travis-ci.org/ekmett/lens/builds/567513619

The jobs seem to take about the same time.

One recent data point is https://github.com/haskell/cabal/pull/5538

> Our `sudo: true` infrastructure are full VMs and are generally faster build environments than the `false` counterpart.

Reply:
> `hvr-ghcjs` neither `nodejs` APT repositories aren't on the whitelist, and there is no way to put them there at the moment

Sigh. I suppose there's nothing we can do about that, then.

In any case, it seems that my knowledge of how Travis works is very out of date. According to https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration, it seems that container-based builds (i.e., ones that use `sudo: false`) have actually been phased out. In fact, all Travis builds have been using VM-based builds since December 2018, and the `sudo` keyword is now deprecated! In that case, I take back my concerns about `sudo: true`—they're entirely unfounded in reality.

This does make me wonder though: do we even need to enable `sudo: true` at all? If I'm reading that blog post correctly, then the behavior of `sudo: true` is always enabled, regardless of configuration.

Reply:
being explicit is better. but you are right, seems sudo is always there now according to https://docs.travis-ci.com/user/reference/overview/

Reply:
I'll remove `sudo: true`.